### DISCUSSION TITLE
Respect type: ignore

### DISCUSSION THREAD

**Discussion by Freed-Wu** [astral-sh/ruff#9937] [Ideas]
```sh
$ pre-commit run -a
ruff..........................................................................Failed
- hook id: ruff
- exit code: 1

binding.gyp:1:1: B018 Found useless expression. Either assign it to a variable or remove it.
Found 1 error.

pyright.......................................................................Failed
- hook id: pyright
- exit code: 1

/home/wzy/Desktop/tree-sitter-m3u/binding.gyp
  /home/wzy/Desktop/tree-sitter-m3u/binding.gyp:1:1 - warning: Expression value is unused (reportUnusedExpression)
```

I must add `  # noqa: B018 # type: ignore` to disable B018 and warning.

Can ruff respect `# type: ignore`? just ignore all warnings in this line.

**Comment by dhruvmanila** [comment]
Hey, thanks for the suggestion. Unfortunately, I don't think that is possible because a `type: ignore` comment is specifically to disable type checking and not linting. The reason Pyright respects the pragma comment is because it is a type checker while Ruff is mainly a linter (and formatter).

**Reference:**
- PEP 484: https://peps.python.org/pep-0484/#compatibility-with-other-uses-of-function-annotations
- Mypy docs: https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#silencing-type-errors
